database/ffldb/blockio.go

```diff
@@ -713,36 +720,61 @@ func (s *blockStore) handleRollback(oldBlockFileNum, oldBlockOffset uint32) {
 }
 
 // scanBlockFiles searches the database directory for all flat block files to
-// find the end of the most recent file.  This position is considered the
-// current write cursor which is also stored in the metadata.  Thus, it is used
-// to detect unexpected shutdowns in the middle of writes so the block files
-// can be reconciled.
-func scanBlockFiles(dbPath string) (int, uint32) {
-	lastFile := -1
-	fileLen := uint32(0)
-	for i := 0; ; i++ {
-		filePath := blockFilePath(dbPath, uint32(i))
+// find the first file, last file, and the end of the most recent file.  The
+// position at the last file is considered the current write cursor which is
+// also stored in the metadata.  Thus, it is used to detect unexpected shutdowns
+// in the middle of writes so the block files can be reconciled.
+func scanBlockFiles(dbPath string) (
+	firstFile int, lastFile int, lastFileLen uint32, err error) {
+
+	files, err := filepath.Glob(filepath.Join(dbPath, "*"+blockFileExtension))
+	if err != nil {
+		return 0, 0, 0, err
+	}
+	sort.Strings(files)
+
+	lastFile, firstFile = -1, -1
+	for i, file := range files {
```

Comment:
Why do we need to actually _scan_ all the files here? With this new usage of `filepath.Glob` we get all the file names up front, then sort them. So we can just get the first, and the last file directly using indexes. Then this way, we don't need to continue scanning and stat'ing them all after we've found the first and last file above.

Reply:
One thing I had in mind was that you could technically have a `511MiB` `.fdb` file if the last block being written onto the disk doesn't fit.

But yeah you'd save a lot of i/o by just skipping. I'll implement that since that seems better overall.